### PR TITLE
labels and job updates to mention new labels

### PR DESCRIPTION
**Pull request by kannon92**
With https://github.com/kubernetes/kubernetes/issues/113714 we are deprecated the unprefixed labels for the job controller. We should add a deprecated note in the website and update examples with the new labels.

### CODE REVIEW

File: content/en/docs/reference/labels-annotations-taints/_index.md
Comment (chrisnegus) [inline content/en/docs/reference/labels-annotations-taints/_index.md:564]:
```suggestion
Kubernetes 1.27 and newer will ignore this label and use the prefixed job-name label.
```

File: content/en/docs/reference/labels-annotations-taints/_index.md
Comment (chrisnegus) [inline content/en/docs/reference/labels-annotations-taints/_index.md:575]:
```suggestion
Kubernetes 1.27 and newer will ignore this label and use the prefixed controller-uid label.
```

**Review by chrisnegus — COMMENTED** [review]
Small suggestion.

File: content/en/docs/concepts/workloads/controllers/job.md
Comment (tengqm) [inline content/en/docs/concepts/workloads/controllers/job.md:202]: Please wrap the long line as we do for other paragraphs?

File: content/en/docs/reference/labels-annotations-taints/_index.md
Comment (tengqm) [inline content/en/docs/reference/labels-annotations-taints/_index.md:589]: Can we add some descriptions to the above two labels?

Comment (tengqm) [reply content/en/docs/reference/labels-annotations-taints/_index.md:589]: By the way, are they labels or annotations?

Comment (kannon92) [reply content/en/docs/reference/labels-annotations-taints/_index.md:589]: Labels. And yep I’ll add some descriptions

Comment (kannon92) [reply content/en/docs/reference/labels-annotations-taints/_index.md:591]: Added.

**Comment by kannon92** [comment]
I’m waiting on https://github.com/kubernetes/kubernetes/pull/114930 to be merged. 

/hold

File: content/en/docs/concepts/workloads/controllers/job.md
Comment (tengqm) [inline content/en/docs/concepts/workloads/controllers/job.md:205]:
```suggestion
In releases prior to 1.27, Jobs used unprefixed labels ie `job-name` and `controller-uid`. 
In 1.27, Job labels will have `batch.kubernetes.io` prefix for `job-name` and `controller-uid`.
```

File: content/en/docs/reference/labels-annotations-taints/_index.md
Comment (tengqm) [inline content/en/docs/reference/labels-annotations-taints/_index.md:673]:
```suggestion
This label is used as a user-friendly way to get Pods corresponding to Jobs.
The job-name comes from the name of the Job and allows for an easy way to get Pods corresponding to the Job.
```

File: content/en/docs/reference/labels-annotations-taints/_index.md
Comment (tengqm) [inline content/en/docs/reference/labels-annotations-taints/_index.md:681]:
```suggestion
This label is used as a programatic way to get all Pods corresponding to the Job.  
```

File: content/en/docs/reference/labels-annotations-taints/_index.md
Comment (tengqm) [inline content/en/docs/reference/labels-annotations-taints/_index.md:682]:
```suggestion
The controller-uid is a unique identifer that gets set in the `selector` field so the Job controller
can get all the corresponding Pods.
```

**Comment by kannon92** [comment]
https://github.com/kubernetes/kubernetes/pull/114930 was merged.

/cancel hold

File: content/en/docs/concepts/workloads/controllers/job.md
Comment (alculquicondor) [inline content/en/docs/concepts/workloads/controllers/job.md:63]: I know these currently show, but since they might eventually delete them, maybe we can remove it from the docs?
Hopefully to reduce their usage

File: content/en/docs/concepts/workloads/controllers/job.md
Comment (alculquicondor) [inline content/en/docs/concepts/workloads/controllers/job.md:163]: This reminds me.... how does `kubectl logs` work for job? If it uses the selector, it might not need to be updated.

File: content/en/docs/concepts/workloads/controllers/job.md
Comment (alculquicondor) [inline content/en/docs/concepts/workloads/controllers/job.md:206]: Probably should start with this line

File: content/en/docs/reference/labels-annotations-taints/_index.md
Comment (alculquicondor) [inline content/en/docs/reference/labels-annotations-taints/_index.md:653]:
```suggestion
Kubernetes 1.27 and newer ignores this label and use the prefixed job-name label.
```

File: content/en/docs/concepts/workloads/controllers/job.md
Comment (kannon92) [reply content/en/docs/concepts/workloads/controllers/job.md:163]: I don't think `kubectl logs` work on the job object.  I think you can only use `kubectl logs` on pods but I would love to be proven wrong.

File: content/en/docs/concepts/workloads/controllers/job.md
Comment (kannon92) [reply content/en/docs/concepts/workloads/controllers/job.md:206]: Not sure what you are asking here.

File: content/en/docs/concepts/workloads/controllers/job.md
Comment (alculquicondor) [reply content/en/docs/concepts/workloads/controllers/job.md:163]: `kubectl logs job/{job-name}` :)

File: content/en/docs/concepts/workloads/controllers/job.md
Comment (alculquicondor) [reply content/en/docs/concepts/workloads/controllers/job.md:206]: The first line of the documentation is what is the current behavior (the behavior for 1.27).
But maybe sig-docs has a different guidance.

File: content/en/docs/concepts/workloads/controllers/job.md
Comment (kannon92) [reply content/en/docs/concepts/workloads/controllers/job.md:63]: I can see that.  Would it cause confusion if someone does `kubectl describe job pi` and the results are different than from the example?

I would vote to have the documentation be the same as what a user would get?

File: content/en/docs/concepts/workloads/controllers/job.md
Comment (kannon92) [reply content/en/docs/concepts/workloads/controllers/job.md:206]: I see.  Daylight savings has not been good on my brain.

File: content/en/docs/concepts/workloads/controllers/job.md
Comment (kannon92) [reply content/en/docs/concepts/workloads/controllers/job.md:163]: Today I learned!  I added that to the doc as I wasn't aware that you could do that.

File: content/en/docs/concepts/workloads/controllers/job.md
Comment (soltysh) [reply content/en/docs/concepts/workloads/controllers/job.md:63]: You can just showcase the one that are relevant and "hide" the ones we want to hide behind `...`  This way we'd "push" users towards using the new labels, still being honest with contents in the docs :wink: 

In general, I agree with Aldo, it's better to push users towards the correct labels than keeping the old deprecated ones. If you look at kubectl, for example, all the deprecated commands are not showing in help. This is similar case, imo.

File: content/en/docs/concepts/workloads/controllers/job.md
Comment (soltysh) [reply content/en/docs/concepts/workloads/controllers/job.md:211]: the prefix is `batch.kubernetes.io/` to be precise :wink:

File: content/en/docs/concepts/workloads/controllers/job.md
Comment (soltysh) [reply content/en/docs/concepts/workloads/controllers/job.md:163]: > This reminds me.... how does `kubectl logs` work for job? If it uses the selector, it might not need to be updated.

I does use a selector, see https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/kubectl/pkg/polymorphichelpers/helpers.go#L174 so that's one of the reasons we will need to support both for quite a while :sweat:

File: content/en/docs/concepts/workloads/controllers/job.md
Comment (kannon92) [reply content/en/docs/concepts/workloads/controllers/job.md:63]: Thank you for the clarification. I’ll update this.

Comment (kannon92) [reply content/en/docs/concepts/workloads/controllers/job.md]: Done.

File: content/en/docs/concepts/workloads/controllers/job.md
Comment (kannon92) [reply content/en/docs/concepts/workloads/controllers/job.md]: Done.

File: content/en/docs/concepts/workloads/controllers/job.md
Comment (alculquicondor) [reply content/en/docs/concepts/workloads/controllers/job.md:155]: Rather not!

Because it uses the selector, we don't need to keep adding the legacy labels not to break kubectl.

Of course, this doesn't apply to out-of-tree controllers using the labels directly.

File: content/en/docs/concepts/workloads/controllers/job.md
Comment (alculquicondor) [inline content/en/docs/concepts/workloads/controllers/job.md:62]:
```suggestion
Annotations:    ...
```

File: content/en/docs/concepts/workloads/controllers/job.md
Comment (alculquicondor) [inline content/en/docs/concepts/workloads/controllers/job.md:205]: I'd remove this.

We don't describe old behavior. And the new labels are already documented as well-known labels.

Comment (kannon92) [reply content/en/docs/concepts/workloads/controllers/job.md]: Done.

File: content/en/docs/concepts/workloads/controllers/job.md
Comment (kannon92) [reply content/en/docs/concepts/workloads/controllers/job.md]: Applied.

**Review by alculquicondor — COMMENTED** [review]
LGTM as approver for job

File: content/en/docs/concepts/workloads/controllers/job.md
Comment (soltysh) [inline content/en/docs/concepts/workloads/controllers/job.md:95]: Nit: 
```
annotations: batch.kubernetes.io/job-tracking: ""
             ...
```

**Review by soltysh — APPROVED** [review]
lgtm from tech pov

File: content/en/docs/reference/labels-annotations-taints/_index.md
Comment (tengqm) [inline content/en/docs/reference/labels-annotations-taints/_index.md:683]:
```suggestion
The `controller-uid` is a unique identifer that gets set in the `selector` field so the Job controller
```

File: content/en/docs/reference/labels-annotations-taints/_index.md
Comment (tengqm) [inline content/en/docs/reference/labels-annotations-taints/_index.md:682]:
```suggestion
This label is used as a programatic way to get all Pods corresponding to a Job.  
```

File: content/en/docs/reference/labels-annotations-taints/_index.md
Comment (tengqm) [inline content/en/docs/reference/labels-annotations-taints/_index.md:673]:
```suggestion
This label is used as a user-friendly way to get Pods corresponding to a Job.
```

File: content/en/docs/reference/labels-annotations-taints/_index.md
Comment (tengqm) [inline content/en/docs/reference/labels-annotations-taints/_index.md:674]:
```suggestion
The `job-name` comes from the `name` of the Job and allows for an easy way to get Pods corresponding to the Job.
```

File: content/en/docs/reference/labels-annotations-taints/_index.md
Comment (tengqm) [inline content/en/docs/reference/labels-annotations-taints/_index.md:664]:
```suggestion
Kubernetes 1.27 and newer ignore this label and use the prefixed `controller-uid` label.
```

File: content/en/docs/reference/labels-annotations-taints/_index.md
Comment (tengqm) [inline content/en/docs/reference/labels-annotations-taints/_index.md:653]:
```suggestion
Kubernetes 1.27 and newer ignore this label and use the prefixed `job-name` label.
```

File: content/en/docs/concepts/workloads/controllers/job.md
Comment (tengqm) [inline content/en/docs/concepts/workloads/controllers/job.md:170]:
```suggestion
Another way to view the logs of a Job:
```

**Review by tengqm — COMMENTED** [review]
/approve

There are some nits in case you want to give it another stab.

**Comment by kannon92** [comment]
> /approve
> 
> There are some nits in case you want to give it another stab.

I went ahead and applied these changes.  Could use a final review!

File: content/en/docs/reference/labels-annotations-taints/_index.md
Comment (divya-mohan0209) [inline content/en/docs/reference/labels-annotations-taints/_index.md:682]: spelling nit.
```suggestion
This label is used as a programmatic way to get all Pods corresponding to a Job.  
```

Comment (kannon92) [reply content/en/docs/reference/labels-annotations-taints/_index.md]: TY. Applied.

**Comment by kannon92** [comment]
@tengqm could I get another lgtm?  I think this is good to go

**Pull request merged**
